### ISSUE TITLE
Docker image can not run as non-root

### ISSUE THREAD

**Issue by julwin** [metabase/metabase#9861]
This is in relation to #4734 - it seems like the container can not run as non-root which is an issue in some kubernetes configurations. 

Switching to UID 2000 or user 'metabase' leads to issues starting up:
![grafik](https://user-images.githubusercontent.com/4594773/56806173-71954080-682b-11e9-9455-4b8bac50a883.png)

:arrow_down: Please click the :+1: reaction instead of leaving a `+1` or `update?` comment

**Comment by camsaul** [comment]
Definitely does not need to be ran as root. Questions like this are best directed to https://discourse.metabase.com

**Comment by camsaul** [comment]
Need to investigate if this is actually still an issue. Originally reported in #4734 and fixed by #4801

@julwin what version of Metabase are you using?

**Comment by camsaul** [comment]
Ok. I was totally mistaken. #4801 creates the new user at runtime and runs Metabase as that user. So it still needs to be launched as root. I will fix this

**Comment by julwin** [comment]
Thanks for looking into this. 😊 Using the latest version docker image btw.

**Comment by eladeyal-intel** [comment]
It tries to created the file in root folder.

**Comment by zheileman** [comment]
Please this is also important for us.
Non-root containers should be the **norm**, not the exception.

**Comment by julwin** [comment]
> 
> 
> Ok. I was totally mistaken. #4801 creates the new user at runtime and runs Metabase as that user. So it still needs to be launched as root. I will fix this

@camsaul Any update on this? Can I help fixing this somehow? I'll gladly assist you in doing this as much as I can..

**Comment by tlrobinson** [comment]
I ran into this trying to deploy our Docker images to Heroku https://devcenter.heroku.com/articles/container-registry-and-runtime#run-the-image-as-a-non-root-user

**Comment by tlrobinson** [comment]
Another issue with our Docker image is Heroku seems to require `CMD` instead of `ENTRYPOINT` in the `Dockerfile`.

For reference, here's how I've been trying to deploy our Docker images to Heroku:

```
heroku apps:create APP_NAME
heroku stack:set container
docker login --username=_ --password=$(heroku auth:token) registry.heroku.com
docker tag LOCAL_IMAGE registry.heroku.com/APP_NAME/web
docker push registry.heroku.com/APP_NAME/web
heroku container:release web
heroku logs -t
```

**Comment by julwin** [comment]
bump?

**Comment by ztsen** [comment]
any solution on this?

**Comment by zheileman** [comment]
Bump?

**Comment by nadilas** [comment]
All issues are marked as duplicate, which is fine but the to redirect to an issue with no activity since over a year? Can we help to get this closed? What is the blocking point currently?

**Comment by flamber** [comment]
@nadilas Have you seen PR #13214? This is not a small change, so we need to carefully review it before committing it.

**Comment by nadilas** [comment]
@flamber apologies, I stand corrected. Let me know if I can assist

**Comment by camsaul** [comment]
Fixed by #13214
